go-controller/pkg/node/route_manager.go

```diff
@@ -47,7 +44,6 @@ func (rm *routeManager) run(stopCh <-chan struct{}) {
 	subscribed, routeEventCh = subscribeNetlinkRouteEvents(stopCh)
 	ticker := time.NewTicker(rm.syncPeriod)
 	defer ticker.Stop()
-	defer rm.wg.Done()
```

Comment:
IIUC, earlier we were passing a waitgroup but never incremented it and were simply deferring the Done call which would try to decrement but there was nothing to decrement right? nice catch

Reply:
yup, that is it